### PR TITLE
[AIRFLOW-6564] Additional diagnostics information on CI check failure

### PR DESCRIPTION
**Pull request by potiuk**
---
Issue link: [AIRFLOW-6564](https://issues.apache.org/jira/browse/AIRFLOW-6564)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by potiuk** [comment]
For now I have not yet tried to improve entropy retrieval from Docker (I will see if that can be achieved easily) but for now I added a bit more sophisticated way of making sure the environment for tests is sane. 

I test each db/integration in a loop n-times + I added the special case for MySQL with parsing the logs + we have full diagnostics in case any of the checks fails. 

Pls. take a look!

File: airflow/jobs/scheduler_job.py [hunk 1]
```diff
@@ -990,7 +990,8 @@ def _change_state_for_tis_without_dagrun(self,
                 .with_for_update() \
                 .all()
             for ti in tis_to_change:
-                ti.set_state(new_state, session=session)
+                ti.set_state(new_state,
```
Comment (Fokko) [inline airflow/jobs/scheduler_job.py]: Unrelated change

File: airflow/utils/db.py [hunk 2]
```diff
@@ -501,3 +501,13 @@ def resetdb():
...
     initdb()
+
+
+@provide_session
+def check(session=None):
+    """
+    Checks if the database works.
+    @param session: session of the sqlalchemy
+    """
+    session.execute('select 1 as is_alive;')
```
Comment (Fokko) [inline airflow/utils/db.py:512]: Does this returns an exit code != 0 if it fails?

**Comment by potiuk** [comment]
I also updated mysql to 5.7 - I might remove that and make a separate PR.

**Review by Fokko — COMMENTED** [review]
Two minor points, apart from that: LGTM

File: airflow/utils/db.py [hunk 3]
```diff
@@ -501,3 +501,13 @@ def resetdb():
     Base.metadata.drop_all(connection)  # pylint: disable=no-member
 
     initdb()
+
+
+@provide_session
+def check(session=None):
+    """
+    Checks if the database works.
+    @param session: session of the sqlalchemy
```
Comment (mik-laj) [inline airflow/utils/db.py]:
```suggestion

    :param session: session of the sqlalchemy
```

File: airflow/jobs/scheduler_job.py (hunk 1 above)
Comment (potiuk) [reply airflow/jobs/scheduler_job.py]: Ah yeah. it was accidental! Thanks!

File: airflow/utils/db.py (hunk 2 above)
Comment (potiuk) [reply airflow/utils/db.py:512]: It throws an exception :). Which in turn exits with non-zero exit code :)

File: airflow/utils/db.py (hunk 3 above)
Comment (potiuk) [reply airflow/utils/db.py]: Right!

File: airflow/bin/cli.py [hunk 4]
```diff
@@ -837,6 +837,12 @@ class CLIFactory:
             'help': "Runs a shell to access the database",
             'args': tuple(),
         },
+        {
+            'func': lazy_load_command('airflow.cli.commands.db_command.check'),
+            'name': 'check',
+            'help': "Chack if the database can be reached.",
+            'args': tuple(),
+        },
```
Comment (kaxil) [inline airflow/bin/cli.py]: We have the `/health` endpoint already, do you think that would cover this and we won't need a new command?

Endpoint: https://github.com/apache/airflow/blob/master/airflow/www/views.py#L182-L210

>An endpoint helping check the health status of the Airflow instance, including metadatabase and scheduler.

File: airflow/utils/db.py [hunk 5]
```diff
@@ -501,3 +501,13 @@ def resetdb():
...
+
+
+@provide_session
+def check(session=None):
+    """
+    Checks if the database works.
+    :param session: session of the sqlalchemy
+    """
+    session.execute('select 1 as is_alive;')
+    log.info("Connection successful.")
```
Comment (kaxil) [inline airflow/utils/db.py:513]: I think the `health` endpoint I mentioned above would be more useful as it checks if the Scheduler can connect to the metadata DB. `session.execute('select 1 as is_alive;')` would check if the machine where the cli is run can execute that query on the Metadb -> i.e. it will check connection between Machine running CLI with DB

Comment (kaxil) [reply airflow/utils/db.py:513]: WDYT? 
If it is just for CI, how about just using curl to access the endpoint?

File: airflow/bin/cli.py (hunk 4 above)
Comment (potiuk) [reply airflow/bin/cli.py]: But we do not run webserver nor airflow at this point yet. It's just environment we check.

File: airflow/bin/cli.py [hunk 6]
```diff
@@ -837,6 +837,12 @@ class CLIFactory:
             'help': "Runs a shell to access the database",
             'args': tuple(),
         },
+        {
+            'func': lazy_load_command('airflow.cli.commands.db_command.check'),
+            'name': 'check',
+            'help': "Chack if the database can be reached.",
```
Comment (kaxil) [inline airflow/bin/cli.py]:
```suggestion
            'help': "Check if the database can be reached.",
```

File: airflow/utils/db.py (hunk 5 above)
Comment (potiuk) [reply airflow/utils/db.py:513]: Again - we do not have neither scheduler nor webserver nor any of the workers. It's just checking if the environment we have is "sane" - i.e. if the databases are there and can be queried, if all the integration for CI tests are reachable at the destination endpoints. 

We explicitly only check database availability before we start tests.

File: airflow/bin/cli.py (hunk 6 above)
Comment (potiuk) [reply airflow/bin/cli.py]: Right!
